gen_proc_macro/src/rc.rs

```diff
@@ -0,0 +1,128 @@
+    if !punct.iter().any(|input| {
+        match input {
+            Pat::Type(arg) => {
+                match &*arg.ty {
+                    Type::Path(ty) => {
+                        ty.path.segments.iter().any(|seg| {
+                            seg.ident
+                                == parse_str::<Ident>("Co").expect("Ident parse failed")
+                        })
+                    }
+                    _ => false,
+                }
+            }
+            _ => false,
+        }
+    }) {
+        let arg =
+            match parse_str::<FnArg>(&format!("co: ::genawaiter::rc::Co<{}>", co_ty)) {
+                Ok(FnArg::Typed(x)) => x,
+                _ => abort_call_site!("string Pat parse failed Co<...>"),
+            };
+        punct.push(Pat::Type(arg))
+    }
+}
+
+/// Parses a `Stmt` to find closure then calls `add_coroutine_arg_cls`
+/// and `replace_yield_cls`.
+pub(crate) fn parse_cls(args: TokenStream, mut function: &mut Stmt) {
+    let co_type = args.to_string();
+    match &mut function {
+        Stmt::Local(loc) => {
+            if let Some((_eq, expr)) = loc.init.as_mut() {
+                match &mut **expr {
+                    Expr::Closure(closure) => {
+                        add_coroutine_arg_cls(&mut closure.inputs, co_type.clone());
+                        replace_yield_cls(&mut *closure.body);
+                    }
+                    Expr::Call(call) => {
+                        for arg in call.args.iter_mut() {
+                            match arg {
+                                Expr::Closure(closure) => {
+                                    add_coroutine_arg_cls(
+                                        &mut closure.inputs,
+                                        co_type.clone(),
+                                    );
+                                    replace_yield_cls(&mut *closure.body);
+                                }
+                                _ => {}
+                            }
+                        }
+                    }
+                    Expr::Unsafe(unsf) => {
+                        for stmt in unsf.block.stmts.iter_mut() {
+                            match stmt {
+                                Stmt::Expr(Expr::Call(call)) => {
+                                    for arg in call.args.iter_mut() {
+                                        match arg {
+                                            Expr::Closure(closure) => {
+                                                add_coroutine_arg_cls(
+                                                    &mut closure.inputs,
+                                                    co_type.clone(),
+                                                );
+                                                replace_yield_cls(&mut *closure.body);
+                                            }
+                                            _ => {}
```

Comment:
I think this is another good place for the visitor pattern. I don't like all these `_ => {}` which ignore malformed input.